iotilesensorgraph/iotile/sg/parser/statements/on_block.py

```diff
@@ -121,7 +121,16 @@ def execute_before(self, sensor_graph, scope_stack):
             else:
                 combiner = '||'
 
-            sensor_graph.add_node(u"({} {} {} {} {}) => {} using copy_latest_a".format(stream_a, trigger_a, combiner, stream_b, trigger_b, trigger_stream))
+            if 'input' in str(stream_a) and 'input' not in str(stream_b):
```

Comment:
Same, check for selector

Reply:
stream_a and stream_b are both DataStream objects so they don't have a selector. I could if the stream_type variable is a DataStream.InputType. 

If this is the case, I will add property function 'input' into the DataStream object that's similar to output:
https://github.com/iotile/coretools/blob/master/iotilesensorgraph/iotile/sg/stream.py#L70

Reply:
Correct, yeah you should add that property